seed/data_importer/tasks.py

```diff
@@ -1363,12 +1277,40 @@ def _map_additional_models(ids, file_pk, progress_key):
     # Don't query the org table here, just get the organization from the import_record
     org = import_file.import_record.super_organization
 
+    from_zipfile = import_file.uploaded_filename.endswith('.zip')
+
     # grab all property states linked to the import file and finish processing the data
     property_states = PropertyState.objects.filter(id__in=ids)
     for property_state in property_states:
         if source_type == BUILDINGSYNC_RAW:
-            # we expect only one buildingfile to be associated with the property
-            building_file = BuildingFile.objects.get(property_state=property_state.id)
+            # create a BuildingFile
+            if from_zipfile:
+                filename = property_state.extra_data.get('_source_filename')
+                if filename is None:
+                    raise Exception('Missing the filename in extra data')
+                with zipfile.ZipFile(import_file.file, 'r', zipfile.ZIP_STORED) as openzip:
+                    new_file = SimpleUploadedFile(
+                        name=filename,
+                        content=openzip.read(filename),
+                        content_type='application/xml')
+            else:
+                filename = import_file.uploaded_filename
+                new_file = SimpleUploadedFile(
+                    name=filename,
+                    content=import_file.file.read(),
+                    content_type='application/xml'
+                )
+
+            if filename == '':
```

Comment:
Added this check because I realized creating a file without a filename would result in the building file's file to be `None`, so when trying to parse the file it'd complain about trying to read a non-existent file